lib/pgBackRest/Config/Config.pm

```diff
@@ -18,2089 +18,103 @@ use pgBackRest::Common::Exception;
-        &OPTION_RULE_ALLOW_LIST =>
-        {
-            lc(OFF)    => true,
-            lc(ERROR)  => true,
-            lc(WARN)   => true,
-            lc(INFO)   => true,
-            lc(DETAIL) => true,
-            lc(DEBUG)  => true,
-            lc(TRACE)  => true
-        },
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD_CHECK => true,
-            &CMD_EXPIRE => true,
-            &CMD_INFO => true,
-            &CMD_RESTORE => true,
-            &CMD_STANZA_CREATE => true,
-            &CMD_STANZA_UPGRADE => true,
-            &CMD_START => true,
-            &CMD_STOP => true
-        }
-    },
-
-    &OPTION_LOG_LEVEL_FILE =>
-    {
-        &OPTION_RULE_SECTION => CONFIG_SECTION_GLOBAL,
-        &OPTION_RULE_TYPE => OPTION_TYPE_STRING,
-        &OPTION_RULE_DEFAULT => OPTION_DEFAULT_LOG_LEVEL_FILE,
-        &OPTION_RULE_ALLOW_LIST =>
-        {
-            lc(OFF)    => true,
-            lc(ERROR)  => true,
-            lc(WARN)   => true,
-            lc(INFO)   => true,
-            lc(DEBUG)  => true,
-            lc(DETAIL) => true,
-            lc(TRACE)  => true
-        },
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD_CHECK => true,
-            &CMD_EXPIRE => true,
-            &CMD_INFO => true,
-            &CMD_RESTORE => true,
-            &CMD_STANZA_CREATE => true,
-            &CMD_STANZA_UPGRADE => true,
-            &CMD_START => true,
-            &CMD_STOP => true
-        }
-    },
-
-    &OPTION_LOG_LEVEL_STDERR =>
-    {
-        &OPTION_RULE_SECTION => CONFIG_SECTION_GLOBAL,
-        &OPTION_RULE_TYPE => OPTION_TYPE_STRING,
-        &OPTION_RULE_DEFAULT => OPTION_DEFAULT_LOG_LEVEL_STDERR,
-        &OPTION_RULE_ALLOW_LIST =>
-        {
-            lc(OFF)    => true,
-            lc(ERROR)  => true,
-            lc(WARN)   => true,
-            lc(INFO)   => true,
-            lc(DETAIL) => true,
-            lc(DEBUG)  => true,
-            lc(TRACE)  => true
-        },
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD_CHECK => true,
-            &CMD_EXPIRE => true,
-            &CMD_INFO => true,
-            &CMD_RESTORE => true,
-            &CMD_START => true,
-            &CMD_STOP => true
-        }
-    },
-
-    &OPTION_LOG_TIMESTAMP =>
-    {
-        &OPTION_RULE_SECTION => CONFIG_SECTION_GLOBAL,
-        &OPTION_RULE_TYPE => OPTION_TYPE_BOOLEAN,
-        &OPTION_RULE_DEFAULT => OPTION_DEFAULT_LOG_TIMESTAMP,
-        &OPTION_RULE_COMMAND =>
-        {
-            &CMD_ARCHIVE_GET => true,
-            &CMD_ARCHIVE_PUSH => true,
-            &CMD_BACKUP => true,
-            &CMD
```

Comment:
This won't always be the case.  The duplicated constants in Config.pm will be moved to the C code and auto-generated -- that should happen before the branch is committed but I may leave it for a future commit.